### ISSUE TITLE
Streamline schedules (do this first)

### ISSUE THREAD

**Issue by Derp7777** [CRLSCSClub/CRLSTime#35]
Right now, we have the clock read from one hard-coded schedule. What we need to do is streamline how it reads the schedule. We will make them lists of nested lists (With each block, e.g. `[[8,05,0],[9,25,0],[9,29,0],etc]`) and have it read from a schedule list where we can change which list it reads from (e.g. `func(list1) --> func(list2)`). Once this is done and we can hard-code what the schedules will be (xblock1, xblock2, xblock3, confrenceEarly (no blocks 1-2), conferenceLate (no blocks lunch-3-4), etc). After this, we will move on to how the schedules will be changed.

**Comment by Derp7777** [comment]
```javascript
function doChangeBS(dontChange) {
var date = new Date();
// dontChange is array of nested arrays with two ints each (eg [3,13] for 3/13)
// dontChange would look like [[3,12],[3,13],etc]
// currentDate would be like one of the dates (like [3,13])
var currentDate = [date.getMonth() + 1, date.getDate()]
if (dontChange.indexOf(currentDate) == -1) {
    return true
} else {
    return false
}

if (doChangeBS(dontChange)) {
    if (blackOrSilver = "Silver") {
        blackOrSilver = "Black")
    } else {
        blackOrSilver = "Silver"
}
```

**Comment by psvenk** [comment]
I agree; there is a lot of redundancy in the code for the schedule. One point on which I disagree with you is that we should have an array like `[[8, 5, 0], [9, 25, 0], ...]`. First, I think that we should include the labels ("morning", "1", "CM", "2", etc.) in the array, and second, I think we can omit the second. Therefore, this is how I would implement `scheduleA`:

```js
/*
  Each schedule variable is an array whose elements are "blocks".
  Each block is an array whose elements are:
    Index 0: label
    Index 1: start time hour
    Index 2: start time minute
  For example, block "LA" (Lunch A) starts at 11:12.
  The end time is inferred from the next block's start time.
*/
var scheduleA = [
  ["morning", 0, 0],
  ["before", 6, 25],
  ["1", 8, 5],
  ["1-CM", 9, 25],
  ["CM", 9, 29],
  ["CM-2", 9, 44],
  ["2", 9, 48],
  ["2-LA", 11, 8],
  ["LA", 11, 12],
  ["LA-3", 11, 42],
  ["3", 11, 46],
  ["3-4", 13, 6],
  ["4", 13, 10],
  ["after", 14, 30],
  ["end", 23, 59]
];
```
Because `getPeriodIndex()` and `getPeriodLength()` both access the `start` property of a period (block), I would implement a function that would return the same object by reading from the array:
```js
function getStart(block) {
  return today.setHours(block[1], block[2], 0, 0);
}
```
This means that instead of `block.start`, we can use `getStart(block)` to get the exact same object, without the memory overhead of storing the schedules in arrays of objects rather than arrays of arrays.

I would implement `getPeriodIndex()` and `getPeriodLength()` like this:
```js
function getPeriodIndex() {
  var now = ServerDate;
  for (i = 0; true; index++) {
    if (now > getStart(schedule[i]) && now < getStart(schedule[i])) {
      return index;
    }
  }
}

function getPeriodLength(i) {
  return (getStart(schedule[i + 1]) - getStart(schedule[i])) / 60000; // in minutes
}
```

Similarly, in `draw.js`, I would change the one line that treats each block as an object ([line 119](https://github.com/CRLSCSClub/CRLSTime/blob/69d32e8ce5133f59652bfcf354f85ff427123e33/draw.js#L119)), and change it from:
```js
var periodLabel = schedule[periodIndex].label;
```
to:
```js
var periodLabel = schedule[periodIndex][0];
```

By the way, I found <https://beautifier.io/> very helpful for writing JavaScript code when I don't have access to a code editor (e.g. Notepad++, SciTe, Emacs).

**Comment by psvenk** [comment]
I plan on implementing and testing these changes a few days to a week from now, and I will open a pull request after everything has been verified to work (including older browsers).

**Comment by psvenk** [comment]
Fixed in #45.
